tests/test_transpiler_optimizer.py

```diff
@@ -7,10 +7,9 @@
 
 
 def star_connectivity():
```

Comment:
This function is defined many times -> you could turn into a fixture, and place it in `conftest.py`.

In principle, you could refactor tests to have a tests subpackage for the transpiler, such that you will be able to use the `conftest.py` scoped in there, rather than the global one.
This I leave up to you.

Reply:
There are different uses of `star_connectivity`. I tried adding this to `conftest.py`, but I could not find a simple function that meets both conditions.

```py
#1
@pytest.mark.parametrize("connectivity", [star_connectivity(), grid_connectivity()])
def test_sabre_random_circuits(n_gates, look, decay, placer, connectivity):

#2
@pytest.mark.parametrize("middle_qubit", [0, 2, 4])
def test_star_router(nqubits, depth, middle_qubit, measurements, unitaries):
    connectivity = star_connectivity(middle_qubit)
```

This version only handles case #2.
```py
@pytest.fixture
def star_connectivity(request):
    middle_qubit = 2 if request.param is None else request.param
    chip = nx.Graph()
    chip.add_nodes_from(list(range(5)))
    graph_list = [(i, middle_qubit) for i in range(5) if i != middle_qubit]
    chip.add_edges_from(graph_list)
    return chip
```
However, since this function will be removed in the transpiler refactor, I will leave it as is for now.